Makefile

```
@@ -46,6 +46,12 @@ build: ## Builds the all docker setup.
 	$(MAKE) install_hooks
 	docker compose build
 
+inactive_issue_checker:
```

Why is this being modified? We aren't using docker in actions any more and I don't see why we need to run docker for something like this anyway.

---

I just followed the convention which is being used in oppia, so in very similar fashion I ran the script using makefile. If it isn't needed anymore then I can definitely choose another way, pls let me know if to do so.

---

What convention are you following exactly? Your GitHub Action should be following the conventions used in .github/workflows/notify_... and I don't see a docker installation step in there.

---

- The convention which I followed after seeing workflows is to hit the inactive issues checker script with the help of  makefile. 
- The reason that why I don't used docker is : there were some dependencies needed in script which wasn't working with docker process at that time. 

Just tell me exactly what I am supposed to change, I'll do it soon.

---

No, I am not going to do that. Take a look at the existing workflows in .github/workflows, understand them, and make sure that your action conforms to the existing patterns.

It is not up to the reviewers to write your code for you. You need to understand what you are doing.

---

Sure here is the way which I thought wihtout using make command in workflow: 


```suggestion

- name: Check inactive issues
      env:
        GITHUB_TOKEN: ${{ secrets.BOT_TOKEN }}
      run: |
        python -m pip install --upgrade pip
        python -m pip install requests
        python scripts/inactive_issue_checker.py
```

Removing command from makefile and replacing `make inactive_issue_checker` with the suggested script.

---

You haven't followed the instructions in my previous comment, so I am going to refer you back to that.

---

Hey @seanlip as per various workflows present in the repository I gotta know about the general patterns as : 
- Checkout repositories 
- Merge develop and set up dependencies.
- Run the job for which these setup is done. 
- And then finally report any failure if occur any. 

Now mistakes/errors which I could notice in my workflow is : 
- Version of checking out repo is v3 while it is v4 in other workflows. 
- No such merge action is present `./.github/actions/merge` while actually it is to be `./.github/actions/merge-develop-and-set-up-dependencies` 
- There is no need of intializing containers as per various workflows (which will finally save time from running this job) 
- And then finally that `make inactive_issue_checker` job, which in place I should use python command to run the script. 

So is that modifications you were talking about or something else is left ?

---

Yes, I'm talking about things like this. Follow the patterns used by existing workflows in the codebase.

---

Done. 
Anything else left @seanlip ?

---

I don't expect to see any Makefile change and I expect the .github/workflows yml files to all be consistent. For example, other "notification" yaml workflows do not have python installation steps.

Please try again.